lib/src/Uncertainty/Algorithm/Optimization/EfficientGlobalOptimization.cxx

```diff
@@ -228,22 +228,15 @@ void EfficientGlobalOptimization::run()
   OptimizationResult result(getProblem());
 
   UnsignedInteger iterationNumber = 0;
+  // use the provided kriging result at first iteration
+  KrigingResult metaModelResult(krigingResult_);
+
+  // EGO expects KrigingAlgorithm to estimate hyperparameters with a single-start OptimizationAlgorithm.
+  const Bool defaultGLMUseMultiStart = ResourceMap::GetAsBool("GeneralLinearModelAlgorithm-DefaultUseMultiStart");
+  ResourceMap::SetAsBool("GeneralLinearModelAlgorithm-DefaultUseMultiStart", false);
```

Comment:
No. ResourceMap is global and should not be changed by other objects unless no other option is available. Here you can use a MultiStart with a sample reduced to the unique starting point. Another option is to add a member to GLMAlgo dedicated to the use of the automatic MultiStart strategy (with the ResourceMap value as default) and use an accessor to this member to change the behaviour of your instance of GLMAlgo. If none of these options fits your needs, at least restore the value of the ResourceMap key as soon as possible to reduce the risk of influencing all the other GLMAlgo instances that may be created (e.g. in another thread) during the time you changed it.

Reply:
I got a little lazy here. The reason I did this was that `GLMAlgo` by default calls TNC and Cobyla's `setIgnoreFailure(true)`. I guess i will have to copy this code into `EfficientGlobalOptimization`.

Reply:
No. You add a new member to the GLMAlgo class and the corresponding accessors, and you use it to deactivate the multistart strategy regardless of the value of the ResourceMap key. Don't copy/paste code as much as possible.

Reply:
OK, I have removed this ugly trick. Thanks.